### PR TITLE
Fix Virginia low-income credit calculation and itemized deductions

### LINKED ISSUE #4260: Fix Virginia low-income credit calculation
Using the 0.711.6 version, the amounts generated for the VA low-income credit are difficult to take seriously.   Below is a test case to which PEUS assigns a low-income credit of almost 1.4 million dollars.

```
- name: Tax unit with taxsimid 41750 in h21.its.csv and h21.ots.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        age: 54
        employment_income: 14_010
        taxable_interest_income: 5_505
        short_term_capital_gains: 2_505
        rent: 20000
      person2:
        age: 54
        employment_income: 1_010
        taxable_interest_income: 5_505
        short_term_capital_gains: 2_505
      person3:
        age: 11
      person4:
        age: 11
      person5:
        age: 16
    tax_units:
      tax_unit:
        members: [person1, person2, person3, person4, person5]
        tax_unit_childcare_expenses: 3_000
    spm_units:
      spm_unit:
        members: [person1, person2, person3, person4, person5]
    households:
      household:
        members: [person1, person2, person3, person4, person5]
        state_fips: 51  # VA
  output:
   va_low_income_tax_credit: 1_395_000
```

### LINKED ISSUE #4311: Fix Virginia itemized deductions calculations
Example from [Schedule A](https://www.tax.virginia.gov/sites/default/files/taxforms/individual-income-tax/2021/schedule-2021.pdf):
image
image

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4260

Fixes #4311

### CODE REVIEW

**Comment by PavelMakarchuk** [comment]
Will also add itemized ded. fix

File: policyengine_us/variables/gov/states/va/tax/income/deductions/va_itemized_deductions.py
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/deductions/va_itemized_deductions.py:1]: This should be more readable now

File: policyengine_us/variables/gov/states/va/tax/income/deductions/va_reduced_itemized_deductions.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Virginia reduced itemized deductions"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://www.tax.virginia.gov/sites/default/files/taxforms/individual-income-tax/2021/schedule-2021.pdf"
+    defined_for = StateCode.VA
+
+    def formula(tax_unit, period, parameters):
+        # The federal limitation on itemized deductions does not apply during the TCJA period
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/va/tax/income/deductions/va_reduced_itemized_deductions.py:14]: Move this `instant_str` piece to the code section where it's used

File: policyengine_us/variables/gov/states/va/tax/income/deductions/va_reduced_itemized_deductions.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/va/tax/income/deductions/va_reduced_itemized_deductions.py:1]: can you shrink this further? still almost 100 lines. parameterizing some of the lists could help

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/va/tax/income/deductions/va_reduced_itemized_deductions.py
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/deductions/va_reduced_itemized_deductions.py:1]: Add the parameters, 
I could shrink the code by removing the comments (22 lines) but it seems like having throughout documentation of the steps, given the complexity, would be very useful. 
I would also not like to separate more computation steps out as it is all one section of the form

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**